modules/apps/friendly-url/friendly-url-web/src/main/java/com/liferay/friendly/url/web/internal/servlet/FriendlyURLServlet.java

```diff
@@ -0,0 +1,334 @@
+					InfoItemPermissionProvider.class, className);
+
+			if (!infoItemPermissionProvider.hasPermission(
+					PermissionCheckerFactoryUtil.create(
+						_portal.getUser(httpServletRequest)),
+					new InfoItemReference(
+						className, _getClassPK(httpServletRequest)),
+					ActionKeys.UPDATE)) {
+
+				_writeJSON(httpServletResponse, JSONUtil.put("success", false));
+			}
+			else {
+				_friendlyURLEntryLocalService.
+					deleteFriendlyURLLocalizationEntry(
+						_getEntryId(httpServletRequest),
+						_getLanguageId(httpServletRequest));
+
+				_writeJSON(httpServletResponse, JSONUtil.put("success", true));
+			}
+		}
+		catch (Exception exception) {
+			_log.error(exception, exception);
+
+			_writeJSON(httpServletResponse, JSONUtil.put("success", false));
+		}
+	}
+
+	@Override
+	protected void doGet(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException, ServletException {
+
+		try {
+			User user = _portal.getUser(httpServletRequest);
+
+			if (user.isDefaultUser() ||
+				!Objects.equals(
+					_getClassName(httpServletRequest),
+					Layout.class.getName())) {
+
+				_writeJSON(httpServletResponse, JSONUtil.put("success", false));
+			}
+			else {
+				_writeJSON(
+					httpServletResponse,
+					_getFriendlyURLEntryLocalizationsJSONObject(
+						httpServletRequest));
+			}
+		}
+		catch (Exception exception) {
+			_log.error(exception, exception);
+
+			_writeJSON(httpServletResponse, JSONUtil.put("success", false));
+		}
+	}
+
+	@Override
+	protected void doPost(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException, ServletException {
+
+		try {
+			String className = _getClassName(httpServletRequest);
+			long classPK = _getClassPK(httpServletRequest);
+
+			InfoItemPermissionProvider infoItemPermissionProvider =
+				_infoItemServiceTracker.getFirstInfoItemService(
+					InfoItemPermissionProvider.class, className);
+
+			if (!infoItemPermissionProvider.hasPermission(
+					PermissionCheckerFactoryUtil.create(
+						_portal.getUser(httpServletRequest)),
+					new InfoItemReference(className, classPK),
+					ActionKeys.UPDATE) ||
+				!className.equals(Layout.class.getName())) {
+
+				_writeJSON(httpServletResponse, JSONUtil.put("success", false));
+			}
+			else {
+				String languageId = _getLanguageId(httpServletRequest);
+
+				FriendlyURLEntryLocalization friendlyURLEntryLocalization =
+					_friendlyURLEntryLocalService.
+						getFriendlyURLEntryLocalization(
+							_getEntryId(httpServletRequest), languageId);
+
+				_layoutLocalService.updateFriendlyURL(
+					_portal.getUserId(httpServletRequest), classPK,
+					friendlyURLEntryLocalization.getUrlTitle(), languageId);
+
+				_writeJSON(httpServletResponse, JSONUtil.put("success", true));
+			}
+		}
+		catch (Exception exception) {
+			_log.error(exception, exception);
+
+			_writeJSON(httpServletResponse, JSONUtil.put("success", false));
```

Comment:
I have doubts with this behavior, to do not add to the JSON some error 
`JSONUtil.put("error", errorMessage);`
instead of put "success" to "false".

I know that we are doing after : 
`httpServletResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST);`

but Maybe we must add some error context to the failure.

I'm only saying this because : this comment
https://issues.liferay.com/browse/LPS-141978?focusedCommentId=2555271&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2555271 and we didn't give any context to the failure.


@boton what do you think?

Reply:
![](https://media0.giphy.com/media/iXQ8SgaMQAgtq/giphy.gif)